pkg/controller/database/rdsinstance/rdsinstance.go

```diff
@@ -161,13 +161,16 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		cr.Status.SetConditions(xpv1.Unavailable())
 	}
 
-	var upToDate bool
-	var diff string
+	upToDate := true
+	diff := ""
```

Comment:
Easier to read would be

```suggestion
	if meta.WasDeleted(cr) {
		return managed.ExternalObservation{}, nil
	}

	var upToDate bool
	var diff string
```

Reply:
good point!

Just added `ResourceExists: true`, else the MR is deleted without sending a deletion-call due to never reach Delete().